src/scale.c

```diff
@@ -71,27 +71,13 @@ avifBool avifImageScale(avifImage * image, uint32_t dstWidth, uint32_t dstHeight
     image->height = dstHeight;
 
     if (srcYUVPlanes[0]) {
-        // Detect integer overflows in libyuv's ScalePlane() and ScalePlane_12() functions. For a
-        // representative example, see the ScalePlaneBox() function. It calls ScaleScape() to set
-        // x, y, dx, and dy. It then does "y += dy" in a loop with dst_height iterations. It may
-        // call ScaleAddCols2_C(), which does "x += dx" in a loop with dst_width iterations. We
-        // simulate the libyuv code and detect if the addition would overflow.
-        //
-        // Divide srcWidth by dstWidth and return as 16.16 fixed point result.
-        const int fixedPointDivWidth = (int)(((int64_t)(srcWidth) << 16) / dstWidth);
-        // The maximum value of the sum when adding fixedPointDivWidth in a loop with dstWidth
-        // iterations.
-        const int64_t maxFixedValueWidth = (int64_t)fixedPointDivWidth * dstWidth;
-        if (maxFixedValueWidth > INT_MAX) {
+        // A simple conservative check to avoid integer overflows in libyuv's ScalePlane() and
+        // ScalePlane_12() functions.
+        if (srcWidth > 16384) {
```

Comment:
Should `dstWidth` be checked too? I did not check libyuv's implementation but it seems likely that the indexing uses the same variable types for dst as src.

Reply:
libyuv's `ScalePlane()` and `ScalePlane_12()` functions have many code paths. I did not inspect all of them. But in the ones I inspected, it turns out that `dstWidth` and `dstHeight` don't need to be checked. I will explain this using the test in the original code.

We compute `maxFixedValueWidth` as follows:

```
    const int fixedPointDivWidth = (int)(((int64_t)(srcWidth) << 16) / dstWidth);
    const int64_t maxFixedValueWidth = (int64_t)fixedPointDivWidth * dstWidth;
```

And then we require `maxFixedValueWidth <= INT_MAX` to avoid integer overflows in libyuv.

It is easy to verify that if `srcWidth <= 32767`, then the requirement  `maxFixedValueWidth <= INT_MAX` will be satisfied for any value of `dstWidth`.

I hope this explanation makes it plausible that `dstWidth` does not need to be checked.

Note: Ideally I should add comments to justify the proposed checks, but I don't know how to explain the checks concisely...

Reply:
Alright, thanks for the detailed explanation. Out of curiosity, did anyone try fixing libyuv?

Reply:
Yes. Frank Barchard (the author of libavif) is looking into fixing libyuv.